js/src/data/api-fetch-middlewares.test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import { createErrorResponseCatcher } from './api-fetch-middlewares';
+import { API_NAMESPACE } from './constants';
+
+describe( 'createErrorResponseCatcher', () => {
+	// Ref: https://github.com/WordPress/gutenberg/blob/%40wordpress/api-fetch%405.1.1/packages/api-fetch/src/index.js#L68-L81
+	const checkStatus = ( response ) => {
+		if ( response.status >= 200 && response.status < 300 ) {
+			return response;
+		}
+
+		throw response;
+	};
+
+	const createFetchHandler = ( status, jsonParsedData ) => () => {
+		const response = {
+			status,
+			json() {
+				return Promise.resolve( jsonParsedData );
+			},
+		};
+		return Promise.resolve( response ).then( checkStatus );
+	};
+
+	const passReject = ( response ) => Promise.reject( response );
+
+	const optionsDefaultParse = { path: `${ API_NAMESPACE }/hi` };
+	const optionsDontParse = { path: `${ API_NAMESPACE }/hi`, parse: false };
```

Comment:
:nail_care: 
Maybe I'm paranoid, but I'd move those into `beforeEach` as we would re-use it in every test

Reply:
Updated by https://github.com/woocommerce/google-listings-and-ads/pull/834/commits/12e78f02d06534727fbcd6945590aa76d40f14bc